### ISSUE TITLE
QueryString paramters templating

### ISSUE THREAD

**Issue by fleed** [ThreeMammals/Ocelot#451]
I'm using version `7.0.7` of the library. I need to transform one route parameter of type Guid into a query parameter.

This is my `ocelot.config`:

```json
{
    "ReRoutes": [
        {
            "DownstreamPathTemplate": "/api/subscriptions/{subscriptionId}/updates?unitId={unitId}",
            "UpstreamPathTemplate": "/api/units/{subscriptionId}/{unitId}/updates",
            "UpstreamHttpMethod": [
                "Get"
            ],
            "DownstreamScheme": "http",
            "DownstreamHostAndPorts": [
                {
                    "Host": "localhost",
                    "Port": 50110
                }
            ]
        }
    ],
    "GlobalConfiguration": {
        "UseServiceDiscovery": false
    }
}
```

I get a *404* response from the actual service. The generated url looks ok from the logs of the actual server, and if I copy&paste the url to a new Postman request, I get a successful response as expected.

Logs by the actual service for the request forwarded by Ocelot (my original GET request: `http://localhost:5003/api/units/50083422-5898-4f21-b938-50a0adb3494b/3bfca484-7316-40b4-8c04-543fed78d7f7/updates`):

```
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET http://localhost:50110/api/subscriptions/50083422-5898-4f21-b938-50a0adb3494b/updates?unitId=3bfca484-7316-40b4-8c04-543fed78d7f7  0
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 348.3264ms
```

If I manually create the request `http://localhost:50110/api/subscriptions/50083422-5898-4f21-b938-50a0adb3494b/updates?unitId=3bfca484-7316-40b4-8c04-543fed78d7f7`, the service replies with `200` as expected.

I see a `  0` at the end of the request in the logs of the actual server, just after the unit id. Could it be the problem?

**Comment by TomPallister** [comment]
@fleed thanks for your interest in Ocelot! I don't think Ocelot supports transforms on the query string atm so this might be a new feature. I will look into it in more detail ASAP.

**Comment by fleed** [comment]
Thank you @TomPallister . For your information, it works if I avoid the query part and do a transformation using route parameters.. anyway, that's not how I planned to design the service api.

**Comment by TomPallister** [comment]
@fleed yeh this is a perfectly reasonable thing to want, Ive just never got round to query string templating or some feature around that. I will have a look at it.

**Comment by TomPallister** [comment]
@fleed Ive made a change that should get your use case working. However query string params won't be supported in UpstreamPathTemplate just yet. This is quite complicated and needs a bit of thought. I will let you know when the change gets released to nuget.

If you want to test it out just pull latest develop of Ocelot!

**Comment by TomPallister** [comment]
Released in 8.0.0

**Comment by fleed** [comment]
Just tested it and it works as expected. Thank you very much!
